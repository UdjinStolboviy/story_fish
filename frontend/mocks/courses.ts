export const response = {
  data: [
    {
      id: 1,
      attributes: {
        header: "Hands-On React. Build advanced React JS Frontend with expert",
        description:
          "**React is the most popular library for building frontend web applications**. Step-by-step by diving into all the basics, I'll introduce you to advanced concepts as well.\n\nWe'll build the minesweeper application from scratch:\n\n-   setup of the development environment\n\n-   configuration of the React JS app\n\n-   basic algorithms of Minesweeper\n\nStep-by-step we'll build the minesweeper game. **During the course, we'll cover the most important topics**.\n\nFirst will be a configuration of the ReactJS application with Typescript and using build tools Webpack+Babel.\n\nSecond, there're presented **TDD or TLD approaches** that I'll try to practice with you during the coding sessions. Probably you don't have enough experience with the tests, but it's ok, there's still a good way to learn it from the course. When you work with code and cover it with test cases, it provides you with guarantees that your code works as you expected. This is the purpose and benefit of the tests.\\\nTools for testing: **Jest, Stryker, React Testing Library**\n\nAlso, I'll try to cover basic **Typescript** concepts and show you a way to improve these skills. During the coding sessions, we'll cover advanced Javascript techniques to make sense of commonly used JS features.\n\nStorybook is the most popular way to build the components library. From our side, it's the most basic part of the course. We'll install and configure Storybook, which will provide us with a full components spec. This approach is called Components Driven Development.\n\n**CSS--IN-JS, emotion, and Styled-Components** are very powerful toolkits to build UI components. Even very complex components styles can be easily produced and supported by it.\n\nWe'll introduce you to **React Hooks** - it's the basic way to make dynamic UI for the applications.\n\nWe'll make a review of **React Router** v5 and also React Router v6.\\\nLast but not least it's **Redux**. Redux is a beautiful way to manage an application state. **We'll use the Redux-Toolkit library** - it simplifies the work with Redux.\\\n**Check out the full curriculum and free preview videos. Join the course risk-free with a 30-day money-back guarantee!**\\\nSee you on the course!",
        createdAt: "2022-04-19T15:19:20.964Z",
        updatedAt: "2022-05-10T11:50:17.393Z",
        publishedAt: "2022-04-19T15:19:43.376Z",
        link: "https://www.udemy.com/course/hands-on-reactjs/",
        subtitle:
          "Practical React with Typescript and Testing Library. React and Storybook. TDD with React Hooks and React + Redux Toolkit",
        cover: {
          data: {
            id: 3,
            attributes: {
              name: "cover4.png",
              alternativeText: "cover4.png",
              caption: "cover4.png",
              width: 1368,
              height: 770,
              formats: {
                thumbnail: {
                  name: "thumbnail_cover4.png",
                  hash: "thumbnail_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 245,
                  height: 138,
                  size: 33.52,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                large: {
                  name: "large_cover4.png",
                  hash: "large_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 1000,
                  height: 563,
                  size: 263.3,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                medium: {
                  name: "medium_cover4.png",
                  hash: "medium_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 750,
                  height: 422,
                  size: 165.98,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                small: {
                  name: "small_cover4.png",
                  hash: "small_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 500,
                  height: 281,
                  size: 89.37,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
              },
              hash: "cover4_dcf1d6a3dc",
              ext: ".png",
              mime: "image/png",
              size: 73.37,
              url: "/covers/hands-on_reactjs_cover.png",
              previewUrl: null,
              provider: "local",
              provider_metadata: null,
              createdAt: "2022-05-09T11:05:18.318Z",
              updatedAt: "2022-05-09T11:05:18.318Z",
            },
          },
        },
      },
    },
    {
      id: 2,
      attributes: {
        header: "TypeScript Basic: Type System, Interfaces, Types, Generics",
        description:
          "**TypeScript** is a programming language developed and maintained by Microsoft. It is a strict syntactical superset of JavaScript and adds optional static typing to the language. TypeScript is designed for the development of large applications and transcompiles to JavaScript. As TypeScript is a superset of JavaScript, existing JavaScript programs are also valid TypeScript programs.\n\nTypeScript may be used to develop JavaScript applications for both client-side and server-side execution (as with Node.js or Deno). There are multiple options available for transcompilation. Either the default TypeScript Checker can be used, or the Babel compiler can be invoked to convert TypeScript to JavaScript.\n\nTypeScript is **JavaScript with syntax for types.**\n\nTypeScript is a strongly typed programming language that builds on JavaScript, giving you better tooling at any scale.\n\nOn the course we quickly create a basic project and then review the most common features\n\n**Who this course is for:**\n\n-   Professional Developers that are looking to keep current by learning the latest programming language of the web development world\n\n-   Backend Developers doing a transition to full stack development\n\n-   Frontend Developers that want to improve their frontend skills\n\n-   Javascript Developers in general that use node as a backend server and would like to learn how to write node in Typescript\n\n-   Javascript Developers looking to learn Typescript",
        createdAt: "2022-04-19T15:20:35.240Z",
        updatedAt: "2022-05-10T11:50:50.670Z",
        publishedAt: "2022-04-19T15:20:38.302Z",
        link: "https://www.udemy.com/course/typescript-in-a-nutshell/",
        subtitle:
          "Learn basic of Typescript, Run Typescript + Webpack and Babel in 2 hour",
        cover: {
          data: {
            id: 4,
            attributes: {
              name: "tslogo.jpg",
              alternativeText: "tslogo.jpg",
              caption: "tslogo.jpg",
              width: 1004,
              height: 462,
              formats: {
                thumbnail: {
                  name: "thumbnail_tslogo.jpg",
                  hash: "thumbnail_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 245,
                  height: 113,
                  size: 3.7,
                  url: "/covers/ts_cover.jpg",
                },
                large: {
                  name: "large_tslogo.jpg",
                  hash: "large_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 1000,
                  height: 460,
                  size: 43.97,
                  url: "/covers/ts_cover.jpg",
                },
                medium: {
                  name: "medium_tslogo.jpg",
                  hash: "medium_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 750,
                  height: 345,
                  size: 25.76,
                  url: "/covers/ts_cover.jpg",
                },
                small: {
                  name: "small_tslogo.jpg",
                  hash: "small_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 500,
                  height: 230,
                  size: 12.03,
                  url: "/covers/ts_cover.jpg",
                },
              },
              hash: "tslogo_91b6c1c0da",
              ext: ".jpg",
              mime: "image/jpeg",
              size: 44.39,
              url: "/covers/ts_cover.jpg",
              previewUrl: null,
              provider: "local",
              provider_metadata: null,
              createdAt: "2022-05-09T11:06:23.764Z",
              updatedAt: "2022-05-09T11:06:23.764Z",
            },
          },
        },
      },
    },
  ],
  meta: {
    pagination: {
      page: 1,
      pageSize: 25,
      pageCount: 1,
      total: 2,
    },
  },
};
