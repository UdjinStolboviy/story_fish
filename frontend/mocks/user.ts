export const mockUser = {
  jwt: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NSwiaWF0IjoxNjUwOTAxNjEwLCJleHAiOjE2NTM0OTM2MTB9.5XCgAR25c3AH_09Jv31CWNF-48mpRqGUat79XGYuNJo",
  user: {
    id: 5,
    username: "John Doe",
    email: "john@doe.com",
    password: "testpass123!",
    provider: "local",
    confirmed: true,
    blocked: false,
    createdAt: "2022-04-20T12:58:41.661Z",
    updatedAt: "2022-04-20T12:58:41.661Z",
  },
};

export const ValidationError = {
  error: {
    status: 400,
    name: "ValidationError",
    message: "Invalid identifier or password",
    details: {},
  },
};

export const RegistrationError = {
  error: {
    status: 400,
    name: "ApplicationError",
    message: "An error occurred during account creation",
    details: {},
  },
};
