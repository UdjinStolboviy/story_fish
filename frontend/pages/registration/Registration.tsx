import type { NextPage } from "next";
import Link from "next/link";
import { useRouter } from "next/router";
import { useSelector, useDispatch } from "react-redux";
import { useForm } from "react-hook-form";
import styled from "@emotion/styled";

import { RootState, AppDispatch } from "@/store";
import {
  selectUser,
  registration,
  RegistrationData,
} from "@/services/userSlice";

import { CenteredTile } from "@/components/Tile";
import { Input, ConditionalFeedback } from "@/components/Input";
import { Button } from "@/components/Button";
import { StyledLink } from "@/components/StyledLink";
import Head from "next/head";
import { Layout } from "@/components/Layout";
import { observer } from "mobx-react-lite";
import { useUserStore } from "@/providers/RootStoreProvider";
import { useState } from "react";

const StyledInput = styled(Input)`
  margin-bottom: 1rem;
`;

const Registration: NextPage = () => {
  const { username, email, jwt, error } = useUserStore();
  const [dataForm, setDataForm] = useState({} as RegistrationData);
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<RegistrationData>();
  const router = useRouter();

  if (Boolean(jwt) && !error) {
    router.push("/user");
  }
  console.log(username, "username");
  useUserStore().registration(dataForm);
  const onSubmit = (data: RegistrationData) => {
    setDataForm(data);
  };

  return (
    <>
      <Head>
        <title>CoursesBox</title>
        <meta name="description" content="IT courses for everyone" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Layout>
        <form onSubmit={handleSubmit(onSubmit)}>
          <CenteredTile header="Create an account">
            <h3>
              <ConditionalFeedback>{error}</ConditionalFeedback>
            </h3>
            <StyledInput
              label="username"
              placeholder="username"
              feedback={
                <ConditionalFeedback>
                  {errors.username?.message}
                </ConditionalFeedback>
              }
              {...register("username", {
                required: "Required field!",
                minLength: { value: 6, message: "Min length 6!" },
                pattern: {
                  value: /^[\w\d\s]+$/,
                  message: "Only letters, numbers and spaces!",
                },
              })}
            />
            <StyledInput
              label="email"
              feedback={
                <ConditionalFeedback>
                  {errors.email?.message}
                </ConditionalFeedback>
              }
              placeholder="email"
              type="email"
              {...register("email", {
                required: "Required field!",
                pattern: {
                  value: /^\S+@\S+$/,
                  message: "Invalid email!",
                },
              })}
            />
            <StyledInput
              label="password"
              type="password"
              role="textbox"
              feedback={
                <ConditionalFeedback>
                  {errors.password?.message}
                </ConditionalFeedback>
              }
              placeholder="password"
              {...register("password", {
                required: "Required field!",
                minLength: { value: 8, message: "Min length 8!" },
              })}
            />
            <Button type="submit">Sign Up</Button>
            <h3>
              <Link href="/login" passHref>
                <StyledLink underline>Login</StyledLink>
              </Link>
            </h3>
          </CenteredTile>
        </form>
      </Layout>
    </>
  );
};

export default observer(Registration);
