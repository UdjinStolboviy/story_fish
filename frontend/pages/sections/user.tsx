import type { NextPage } from "next";
import { useRouter } from "next/router";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";

import { RootState, AppDispatch } from "@/store";
import { selectUser, logout } from "@/services/userSlice";

import { CenteredTile } from "@/components/Tile";
import { Button } from "@/components/Button";

const User: NextPage = () => {
  const router = useRouter();
  const { username, email, error } = useSelector<RootState, RootState["user"]>(
    selectUser
  );
  const dispatch = useDispatch<AppDispatch>();

  useEffect(() => {
    if (!username || Boolean(error)) {
      dispatch(logout());
      router.push("/login");
    }
  }, []);

  const logoutHandler = () => {
    dispatch(logout());
    router.push("/");
  };

  return username && email ? (
    <CenteredTile header="Profile">
      <h3>username: {username}</h3>
      <h3>email: {email}</h3>
      <Button onClick={logoutHandler}>Logout</Button>
    </CenteredTile>
  ) : null;
};

export default User;
