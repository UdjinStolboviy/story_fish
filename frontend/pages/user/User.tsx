import type { NextPage } from "next";
import { useRouter } from "next/router";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";

import { RootState, AppDispatch } from "@/store";
import { selectUser, logout, UserState } from "@/services/userSlice";

import { CenteredTile } from "@/components/Tile";
import { Button } from "@/components/Button";

import Head from "next/head";
import { Layout } from "@/components/Layout";

import { useState } from "react";
import { clearUserInfoFromLocalStorage } from "@/utils/utils";
import { useQueryClient } from "react-query";

const User: NextPage = () => {
  const router = useRouter();
  const [user, setUser] = useState<UserState | null>(null);
  const queryClient = useQueryClient();
  console.log("user ----------------  s ", user);
  useEffect(() => {
    const user: UserState | undefined = queryClient.getQueryData("user");

    if (!user) {
      router.push("/login");
    } else {
      setUser(user);
    }
  }, []);

  const logoutHandler = () => {
    clearUserInfoFromLocalStorage();
    router.push("/");
  };

  return (
    <>
      <Head>
        <title>CoursesBox</title>
        <meta name="description" content="IT courses for everyone" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Layout>
        {user && user.username && user.email ? (
          <CenteredTile header="Profile">
            <h3>username: {user.username}</h3>
            <h3>email: {user.email}</h3>
            <Button onClick={logoutHandler}>Logout</Button>
          </CenteredTile>
        ) : null}
      </Layout>
    </>
  );
};

export default User;
