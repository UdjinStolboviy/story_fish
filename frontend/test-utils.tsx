import { FC, ReactElement } from "react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import { rootReducer, RootState } from "./store";
import { Themes } from "./styles/themes";
import { Layout } from "./components/Layout";

type Options = { preloadedState?: RootState } & RenderOptions;

const customRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

const pageRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <Layout>{children}</Layout>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, pageRender };
