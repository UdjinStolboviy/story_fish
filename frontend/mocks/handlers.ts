import { rest } from "msw";

import { LoginData, RegistrationData } from "@/services/userSlice";

import { mockUser, ValidationError, RegistrationError } from "./user";

const api_url = process.env.NEXT_PUBLIC_STRAPI_API_URL;

export const handlers = [
  rest.post<LoginData>(`${api_url}/auth/local`, (req, res, ctx) => {
    const data = req.body;

    if (
      data.identifier === mockUser.user.email &&
      data.password === mockUser.user.password
    ) {
      return res(ctx.status(200), ctx.json(mockUser));
    }

    return res(ctx.status(400), ctx.json(ValidationError));
  }),

  rest.get(`${api_url}/users/me`, (req, res, ctx) => {
    if (req.headers.get("Authorization") === `Bearer ${mockUser.jwt}`) {
      return res(ctx.status(200), ctx.json(mockUser.user));
    }

    return res(ctx.status(400), ctx.json(ValidationError));
  }),

  rest.post<RegistrationData>(
    `${api_url}/auth/local/register`,
    (req, res, ctx) => {
      const data = req.body;

      if (
        data.email === mockUser.user.email &&
        data.password === mockUser.user.password &&
        data.username === mockUser.user.username
      ) {
        return res(ctx.status(200), ctx.json(mockUser));
      }

      return res(ctx.status(400), ctx.json(RegistrationError));
    }
  ),
];
