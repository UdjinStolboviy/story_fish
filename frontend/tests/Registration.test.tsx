import { render, screen, act, waitFor } from "@/test-utils";
import userEvent from "@testing-library/user-event";
import { useRouter } from "next/router";

import { mockUser } from "@/mocks/user";

import Registration from "@/pages/registration";

jest.mock("next/router", () => ({
  ...jest.requireActual("next/router"),
  useRouter: jest.fn().mockReturnValue({
    query: {},
    push: jest.fn(),
  }),
}));

describe("Registration page", () => {
  it("Render check", () => {
    const { container } = render(<Registration />);
    expect(container).toMatchSnapshot();
  });
  it("Client validation check", async () => {
    render(<Registration />);

    const submitButton = screen.getByRole("button", { name: "Sign Up" });

    await act(async () => {
      userEvent.click(submitButton);
    });

    expect(screen.getAllByText("Required field!")).toHaveLength(3);

    await act(async () => {
      userEvent.type(
        screen.getByRole("textbox", { name: "username Required field!" }),
        "test"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "email Required field!" }),
        "test"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "password Required field!" }),
        "test"
      );
    });

    expect(screen.getByText("Min length 6!")).toBeInTheDocument();
    expect(screen.getByText("Invalid email!")).toBeInTheDocument();
    expect(screen.getByText("Min length 8!")).toBeInTheDocument();

    await act(async () => {
      userEvent.type(
        screen.getByRole("textbox", { name: "username Min length 6!" }),
        "test--!"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "email Invalid email!" }),
        "test@test"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "password Min length 8!" }),
        "testtest!"
      );
    });

    expect(
      screen.getByText("Only letters, numbers and spaces!")
    ).toBeInTheDocument();

    await act(async () => {
      const usernameInput = screen.getByRole("textbox", {
        name: "username Only letters, numbers and spaces!",
      });
      userEvent.clear(usernameInput);
      userEvent.type(usernameInput, "testtesttest");
    });

    const alerts = screen.getAllByRole("alert");

    expect(alerts).toHaveLength(3);
    expect(alerts[0]).toMatchSnapshot();
    expect(alerts[1]).toMatchSnapshot();
    expect(alerts[2]).toMatchSnapshot();
  });

  it("Server validation error check", async () => {
    render(<Registration />);

    const submitButton = screen.getByRole("button", { name: "Sign Up" });

    act(() => {
      userEvent.type(
        screen.getByRole("textbox", { name: "username" }),
        "testtest"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "email" }),
        "test@test.test"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "password" }),
        "testtest!"
      );
      userEvent.click(submitButton);
    });

    expect(
      await screen.findByText("An error occurred during account creation")
    ).toBeInTheDocument();
  });
  it("Successful registration check", async () => {
    // Mock the router
    const push = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push,
    });

    render(<Registration />);

    const submitButton = screen.getByRole("button", { name: "Sign Up" });

    act(() => {
      userEvent.type(
        screen.getByRole("textbox", { name: "username" }),
        mockUser.user.username
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "email" }),
        mockUser.user.email
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "password" }),
        mockUser.user.password
      );
      userEvent.click(submitButton);
    });

    // Check if the user is redirected to the user page
    await waitFor(() => expect(push).toHaveBeenCalledWith("/user"));
  });
});
