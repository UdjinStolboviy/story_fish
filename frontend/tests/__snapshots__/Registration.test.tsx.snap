// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page Client validation check 1`] = `
<span
  class="css-1reg724"
  role="alert"
>
   
</span>
`;

exports[`Registration page Client validation check 2`] = `
<span
  class="css-1reg724"
  role="alert"
>
   
</span>
`;

exports[`Registration page Client validation check 3`] = `
<span
  class="css-1reg724"
  role="alert"
>
   
</span>
`;

exports[`Registration page Render check 1`] = `
<div>
  <form>
    <div
      class="css-qbqlvy"
    >
      <section
        class="css-6q0t2w"
      >
        <h2>
          Create an account
        </h2>
        <h3>
           
        </h3>
        <label
          class="css-58qgct"
          height="7"
          width="20"
        >
          <span
            class="css-1grri41"
          >
            username
          </span>
          <div
            class="css-52ovgc"
          >
            <input
              class="css-1bj11r8"
              name="username"
              placeholder="username"
            />
          </div>
          <span
            class="css-1reg724"
            role="alert"
          >
             
          </span>
        </label>
        <label
          class="css-58qgct"
          height="7"
          width="20"
        >
          <span
            class="css-1grri41"
          >
            email
          </span>
          <div
            class="css-52ovgc"
          >
            <input
              class="css-1bj11r8"
              name="email"
              placeholder="email"
              type="email"
            />
          </div>
          <span
            class="css-1reg724"
            role="alert"
          >
             
          </span>
        </label>
        <label
          class="css-58qgct"
          height="7"
          width="20"
        >
          <span
            class="css-1grri41"
          >
            password
          </span>
          <div
            class="css-52ovgc"
          >
            <input
              class="css-1bj11r8"
              name="password"
              placeholder="password"
              role="textbox"
              type="password"
            />
          </div>
          <span
            class="css-1reg724"
            role="alert"
          >
             
          </span>
        </label>
        <button
          class="css-1rttvxo"
          color="primary"
          type="submit"
        >
          Sign Up
        </button>
        <h3>
          <a
            class="css-pm0vza"
            href="/login"
          >
            Login
          </a>
        </h3>
      </section>
    </div>
  </form>
</div>
`;
